lib/spack/spack/solver/asp.py

```diff
@@ -1011,35 +1030,83 @@ def package_compiler_defaults(self, pkg):
             )
 
     def package_requirement_rules(self, pkg):
+        rules = self.requirement_rules_from_package_py(pkg)
+        rules.extend(self.requirement_rules_from_packages_yaml(pkg))
+        self.emit_facts_from_requirement_rules(rules)
+
+    def requirement_rules_from_package_py(self, pkg):
+        rules = []
+        for requirements, conditions in pkg.requirements.items():
+            for when_spec, policy, message in conditions:
+                rules.append(
+                    RequirementRule(
+                        pkg_name=pkg.name,
+                        policy=policy,
+                        requirements=requirements,
+                        kind=Provenance.PACKAGE_PY,
```

Comment:
(bug) This should be a `RequirementKind`

Reply:
Fixed in 3eeb585